pkg/kotsadmsnapshot/backup.go

```diff
@@ -254,84 +280,90 @@ func CreateInstanceBackup(ctx context.Context, cluster *downstreamtypes.Downstre
+			return "", errors.Wrapf(err, "failed to create temp dir for app %s", a.Slug)
 		}
 		defer os.RemoveAll(archiveDir)
 
 		err = store.GetStore().GetAppVersionArchive(a.ID, parentSequence, archiveDir)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get app version archive for app %s", a.Slug)
+			return "", errors.Wrapf(err, "failed to get app version archive for app %s", a.Slug)
 		}
 
 		kotsKinds, err := kotsutil.LoadKotsKinds(archiveDir)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to load kots kinds from path")
+			return "", errors.Wrap(err, "failed to load kots kinds from path")
 		}
 
 		backupSpec, err := kotsKinds.Marshal("velero.io", "v1", "Backup")
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to get backup spec from kotskinds")
+			return "", errors.Wrap(err, "failed to get backup spec from kotskinds")
 		}
 
+		var veleroBackup *velerov1.Backup
 		if backupSpec == "" {
-			continue
-		}
-
-		appsSequences[a.Slug] = parentSequence
-		appVersions[a.Slug] = kotsKinds.Installation.Spec.VersionLabel
-
-		renderedBackup, err := helper.RenderAppFile(a, nil, []byte(backupSpec), kotsKinds, kotsadmNamespace)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to render backup")
-		}
-		veleroBackup, err := kotsutil.LoadBackupFromContents(renderedBackup)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to load backup from contents")
-		}
-
-		// ** merge app backup info ** //
-		// included namespaces
-		includedNamespaces = append(includedNamespaces, veleroBackup.Spec.IncludedNamespaces...)
-		includedNamespaces = append(includedNamespaces, kotsKinds.KotsApplication.Spec.AdditionalNamespaces...)
+			// If this is Embedded Cluster, backups are always enabled and we will use a default
+			// minimal backup spec
+			if util.IsEmbeddedCluster() {
+				veleroBackup = getDefaultEmbeddedClusterBackupSpec()
+			} else {
+				continue
+			}
+		} else {
+			renderedBackup, err := helper.RenderAppFile(a, nil, []byte(backupSpec), kotsKinds, kotsadmNamespace)
+			if err != nil {
+				return "", errors.Wrap(err, "failed to render backup")
+			}
+			veleroBackup, err = kotsutil.LoadBackupFromContents(renderedBackup)
+			if err != nil {
+				return "", errors.Wrap(err, "failed to load backup from contents")
+			}
 
-		// excluded namespaces
-		excludedNamespaces = append(excludedNamespaces, veleroBackup.Spec.ExcludedNamespaces...)
+			// If this is EC and it is using the new improved DR (there is a backup and restore custom
+			// resource), dont merge the backup spec.
+			if usesImprovedDR(kotsKinds) {
+				if len(apps) > 0 {
+					return "", errors.New("cannot create backup for Embedded Cluster with multiple apps")
+				}
+				appVeleroBackup = veleroBackup
+				appVeleroBackup.Name = ""
+				appVeleroBackup.GenerateName = a.Slug + "-"
+				appVeleroBackup.Namespace = kotsadmVeleroBackendStorageLocation.Namespace
```

Comment:
do we need to add these?

Reply:
probably not since the namespace is already used when creating the backup below

Reply:
so we dont need the namespace? or we dont need all three?

Reply:
i removed namespace but kept the other two